EIPS/eip-165.md

```diff
@@ -0,0 +1,230 @@
+## Preamble
+
+```
+EIP: <to be assigned>
+Title: ERC-165 Standard Interface Detection
+Author: Christian Reitwießner <chris@ethereum.org>, Nick Johnson <nick@ethereum.org>, RJ Catalano <rj@monax.io>, Fabian Vogelsteller <fabian@frozeman.de>, Hudson Jameson <hudson@hudsonjameson.com>, Jordi Baylina <jordi@baylina.cat>, Griff Green <saveoursonics@gmail.com>, Konrad Feldmeier <konrad.feldmeier@brainbot.com>, William Entriken <github.com@phor.net>
+Type: Standard Track
+Category: ERC
+Status: Draft
+Created: 2018-01-23
+```
+
+## Simple Summary
+
+Creates a standard method to publish and detect what interfaces a smart contract implements.
+
+## Abstract
+
+Herein, we standardize the following:
+
+1. How interfaces are identified
+2. How a contract will publish the interfaces it implements
+3. How to detect if a contract implements ERC-165
+4. How to detect if a contract implements any given interface
+
+## Motivation
+
+For some "standard interfaces" like [the ERC-20 token interface](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20-token-standard.md), it is sometimes useful to query whether a contract supports the interface and if yes, which version of the interface, in order to adapt the way in which the contract is to be interfaced with. Specifically for ERC-20, a version identifier has already been proposed. This proposal stadardizes the concept of interfaces and standardizes the identification (naming) of interfaces.
+
+## Specification
+
+### How Interfaces are Identified
+
+For this standard, an *interface* is a set of [function selectors as calculated in Solidity](http://solidity.readthedocs.io/en/develop/abi-spec.html#function-selector). This a subset of [Solidity's concept of interfaces](http://solidity.readthedocs.io/en/develop/abi-spec.html) and the  `interface` keyword definition which also define return types, mutability and events.
```

Comment:
Note that function selectors are not specific to Solidity, they are part of the ABI specification.

Reply:
Updated: "function selectors as defined by the Ethereum ABI"